apps/language_server/test/experimental/provider/code_action/replace_with_underscore_test.exs

```diff
@@ -71,12 +70,24 @@ defmodule ElixirLS.LanguageServer.Experimental.Provider.CodeAction.ReplaceWithUn
 
   test "produces no actions if the name or variable is not found" do
     assert {_, action} = code_action("other_var = 6", "/project/file.ex", 1, "not_found")
-    assert [] = apply(action)
+
+    source_file = action.source_file
+    diagnostics = get_in(action, [:context, :diagnostics])
+
+    Enum.each(diagnostics, fn diagnostic ->
+      assert [] = ReplaceWithUnderscore.apply(source_file, diagnostic)
+    end)
```

Comment:
why this change? 
Placing asserts in a loop makes them very hard to debug and reason about.

Reply:
I still wanted to test only `ReplaceWithUnderscore.apply` (instead of `CodeAction.handle`) here and that required having both `source_file` and `diagnostics`. I'm looping to call `apply` for every diagnostic

Reply:
I'm confused, this didn't call `CodeAction.handle`

Reply:
I will change it back anyway because of https://github.com/elixir-lsp/elixir-ls/pull/776#discussion_r1138573992